Add sphinx docs

c24t opened the pull request:
This PR adds sphinx boilerplate for generating API docs. Most files are generated with `sphinx-quickstart` and `sphinx-apidoc` with few changes.

This branch builds on https://github.com/open-telemetry/opentelemetry-python/pull/11, I'll rebase as it's updated.

You can build the html docs from this branch as:

```sh
pip install sphinx sphinx_autodoc_typehints sphinx_rtd_theme
sphinx-quickstart
sphinx-apidoc --force -e -o docs opentelemetry
(cd docs; make html)
```

I also pushed the generated html docs to the [`sphinx-docs-built` branch](https://github.com/open-telemetry/opentelemetry-python/tree/sphinx-docs-built/docs/_build/html), which you can check out to view the docs without building them yourself.

Oberon00 commented on docs/opentelemetry.api.rst:
```diff
@@ -0,0 +1,17 @@
+---------------
+
+.. automodule:: opentelemetry.api
+    :members:
```
Since these options seem to be the same everywhere, they should be added to the configuration https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_default_options

c24t commented:
Should we (1) document the constructor args in the class docstring as opencensus does or (2) whitelist `__init__` and keep the docs there? 

Option 1:

![Screen Shot 2019-06-20 at 11 54 07 AM](https://user-images.githubusercontent.com/43799105/59874276-c1d3df00-9352-11e9-80a0-c883e0953b6d.png)

and option 2:

![Screen Shot 2019-06-20 at 11 54 42 AM](https://user-images.githubusercontent.com/43799105/59874286-c7312980-9352-11e9-8f13-db08aefb8218.png)

I have a slight preference for option 1, but option 2 is more explicit.

c24t commented:
I updated the [`sphinx-docs-built` branch](https://github.com/open-telemetry/opentelemetry-python/tree/sphinx-docs-built/docs/_build/html), see the generated html docs there.

c24t commented:
I update the docs for the package move in #21 and removed a lot of the generated API doc boilerplate. After the package move autodoc generates empty docs for the new empty packages from #19, I removed these by hand.

I also moved the `opentelemetry.trace` module docs up to the top-level TOC with the expectation that we'll curate the TOC ourselves instead of leaving it up to autodoc. These changes mean we can't just run `sphinx-apidoc` and `make html` to generate the docs; we have to prune some generated files by hand.

![Screen Shot 2019-06-21 at 12 27 05 PM](https://user-images.githubusercontent.com/43799105/59946649-04afb880-9420-11e9-8f8e-e10cda7c73fd.png)

![Screen Shot 2019-06-21 at 12 27 17 PM](https://user-images.githubusercontent.com/43799105/59946654-07aaa900-9420-11e9-93ee-9ca8c995299c.png)
